dataColl/db.py

```diff
@@ -51,6 +52,47 @@ def query_db(query, args=(), one=False):
     cur.close()
     return (rv[0] if rv else None) if one else rv
 
+def process_form(form_data: dict) -> None:
+    
+    top_level_fields = {}
+    top_level_fields["start_date"] = form_data['time_period']['start']
+    top_level_fields["end_date"] = form_data['time_period']['end']
+    top_level_fields["name_ddpo"] = form_data['name_ddpo']
+    top_level_fields["borough_covered"] = ', '.join([str(elem) for elem in form_data['borough_covered']])
+    top_level_fields["ddpo_short_paragraph"] = form_data["ddpo_short_paragraph"]
+    top_level_fields["casestudy_highlighting_emotional_impact"] = form_data["casestudy_highlighting_emotional_impact"]
+    top_level_fields["casestudy_highlighting_positive_outcome"] = form_data["casestudy_highlighting_positive_outcome"]
+
+    fields = top_level_fields.keys()
+    question_marks = ", ".join("?" * len(fields))
+    column_str = ", ".join([str(elem) for elem in fields])
+
+    sql = f"INSERT into form_data ({column_str}) VALUES ({question_marks})"
+
+    db = get_db()
+    db.execute(sql, ([top_level_fields[key] for key in fields]))
+    db.commit()
+        
+def process_question(form_data: dict, question: str) -> None:
+    
+    if question == "time_period":
+        return None
+    if not isinstance(form_data[question], dict):
+        return None
+
+    keys = form_data[question].keys()
+    question_marks = ", ".join("?" * len(keys))
+    keys_str = ", ".join([str(elem) for elem in form_data[question].keys()])
+
+    sql = f"INSERT into {question} ({keys_str}) VALUES ({question_marks})"
+
+    try:
+        db = get_db()
+        db.execute(sql, ([form_data[question][key] for key in keys]))
+        db.commit()
+    except OperationalError as e:
+        print(e.args)
```

Comment:
We probably need to plumb in some logging, I'm not sure this print will pull through. This would also make the commit fail silently, which isn't good. I suggest that this should result in a JSend-compliant error response being sent to the client.

Reply:
yeah, that was there for debugging, it was alte when I pushed it and didn't think that it would make it fail silently, whoops 😆  I have changed it so that all the errors are captured and send in the json response